### DISCUSSION TITLE
Welcoming 2 new co-maintainers ❤️

### DISCUSSION THREAD

**Discussion by yajo** [copier-org/copier#1180] [Show and tell]
As discussed long time ago with @sisp, I added him today as a co-maintainer. Welcome on board! 🚢 

I also took the chance to add @pawamoy. I didn't ask you, but if you don't want I can remove you. You choose. 😊 

Both are mentioned in the Copier hall of fame (the README footer, basically 😁) and have been very supportive for a long time. I think both of them have deep knowledge about the inner stuff in Copier. I'd still like to keep the hands on the wheel regarding product design, but honestly my free time for Copier is going down and I mostly maintain it from my phone these days, so I think it's time to train new maintainers, just like it happened to me years ago.

I added a new rule to PRs: they need one approval to merge. Thus, if @sisp opens one and @pawamoy approves it (or vice-versa), you can merge those. I'm excluding myself from that rule, just for the sake of speed (when I have some time to develop Copier, I wanna go as fast as I can, to make it more profitable). You should be able to approve and merge my PRs, triage issues, close old PRs, etc.

Currently, the project only has donations for myself in my personal profile. Current donations are modest, but they help a lot in keeping the project alive! For the sake of transparency, @sisp already told me he doesn't want any economic share because the hours he dedicates to Copier are already paid from his employer. @pawamoy, I didn't ask you, but if you feel it's more fair, we can try to set up something more suited for a community, such as community github donations or open collective, etc. Just tell me what you think please.

If needed, anybody can ping us with @copier-org/maintainers now.

Thanks for the help you've given to Copier (and me) until today. Welcome!

**Comment by pawamoy** [comment]
Thanks for ensuring the project's maintainability @yajo!
And thanks for giving me more access :slightly_smiling_face:

About Copier's funding: it's fine by me if you keep getting the donations personally (through your GitHub account). You're at the wheel after all! And I have a sponsorware strategy setup for my own projects :slightly_smiling_face:

**Comment by sisp** [comment]
Thanks for entrusting me with more permissions on this fantastic project, @yajo! :pray: :bow: I'm honored! :blush: And thanks for taking such good care of Copier! :heart: :slightly_smiling_face: I hope the project will continue to benefit from your deep expertise and knack for keeping things simple and for finding pragmatic solutions, as proven again recently regarding the issue on updating a project with secret questions.

In my opinion, Copier hasn't realized its full potential yet, mainly &ndash; I guess &ndash; because this kind of "meta development" (is there a better word for it? :shrug: :laughing:) isn't so well known yet, perhaps because most alternatives don't support the entire project lifecycle but only generate the initial scaffold. I'd love to see Copier change the way developers manage and maintain their projects, and thus also contribute to improving (i.e. consolidating, harmonizing, possibly fixing) project and software quality.

Perhaps slightly off-topic for this discussion, but have you (@yajo and @pawamoy) ever attended PyCon or similar? I imagine talking about Copier at such an event could give it additional spin. :thinking: :smile:

**Reply by yajo** [reply]
> have you (@yajo and @pawamoy) ever attended PyCon or similar?

Nope. I sometimes go to [OCA days](https://odoo-community.org/event/oca-days-2023-liege-2023-11-06-2023-11-07-143/page/introduction-oca-days-2022-liege-copy), but that's not a general-purpose Python event.

**Reply by pawamoy** [reply]
Never did either, but I'd love to go to PyCon in Paris one day (closest one to where I live). Just to attend though, I'm not a speaker haha (at least, yet)

**Comment by sisp** [comment]
> I added a new rule to PRs: they need one approval to merge. Thus, if @sisp opens one and @pawamoy approves it (or vice-versa), you can merge those. I'm excluding myself from that rule, just for the sake of speed (when I have some time to develop Copier, I wanna go as fast as I can, to make it more profitable). You should be able to approve and merge my PRs, triage issues, close old PRs, etc.

@yajo I'm not able to merge a PR (e.g. #1201) when, e.g., @pawamoy has approved it:

![image](https://github.com/copier-org/copier/assets/2206639/c49b00f0-ad23-41b9-b578-5b117cc7525c)

I think this isn't the idea because I believe neither of us is able to merge it; I can't and since @pawamoy approved it, I think he can't either.

**Reply by pawamoy** [reply]
I confirm, cannot merge, same message as @sisp.
> Merging is blocked
The base branch restricts merging to authorized users. [Learn more about protected branches.](https://docs.github.com/articles/about-protected-branches/)

**Reply by yajo** [reply]
Oops... could you try now?

**Reply by sisp** [reply]
Looks good now. GitHub allows enabling auto-merge now on PRs without approval and states that 1 approval is required for merging.

**Reply by yajo** [reply]
Good, that was the goal 😊

**Reply by sisp** [reply]
@yajo It seems I can merge a PR that I've approved myself without another approval from you or @pawamoy. For instance, I've approved the most recent Dependabot PRs and could click on "merge" now. It seems we have too many permissions now.

**Reply by sisp** [reply]
To clarify, an approval is still required, but I can approve a PR and merge it _myself_ afterwards.

**Reply by yajo** [reply]
OK, it should be fixed now, thanks for reporting!

**Reply by sisp** [reply]
So far so good, 2 approving reviews are needed now, only 1 of which I can give myself. I assume GitHub allows you to exempt yourself from the rule. :+1:

**Reply by sisp** [reply]
@yajo I think the current setting is wrong but the previous one was actually correct. Now, when I approve a PR, e.g. by @pawamoy, another approval is needed before we can merge it. Before, I said I could approve and merge a PR; that was for a Dependabot PR though. There, I think it makes sense that one of us can approve and merge it since it's the same as reviewing each others' PRs and merging them. Since I can't approve my own PRs, one of you needs to approve before I can merge it. So a second opinion is (or rather, was) required for our own PRs. Sorry for the confusion. 🙁

**Reply by yajo** [reply]
Thanks! It's one approval required now. That's correct now?

**Reply by sisp** [reply]
Looks good, thanks! :+1:

**Comment by sisp** [comment]
Just a quick note about Copier's popularity: We've crossed the 2000 GitHub ⭐ threshold! 🎉

**Reply by yajo** [reply]
Woah!

I also noticed we're one of the featured 6 flakes in https://flakehub.com:

![1000016786](https://github.com/user-attachments/assets/a6bdcc9c-e416-4d7a-8847-d7a0091c410d)
